extensions/iam/decentralized-identity/identity-did-service/src/main/java/org/eclipse/dataspaceconnector/identity/DecentralizedIdentityService.java

```diff
@@ -88,7 +87,9 @@ public Result<ClaimToken> verifyJwtToken(TokenRepresentation tokenRepresentation
             PublicKeyWrapper publicKeyWrapper = KeyConverter.toPublicKeyWrapper(publicKeyJwk, publicKey.get().getId());
 
             monitor.debug("Verifying JWT with public key...");
-            if (!VerifiableCredentialFactory.verify(jwt, publicKeyWrapper)) {
+            Result<Void> verified = VerifiableCredentialFactory.verify(jwt, publicKeyWrapper, audience);
+            if (verified.failed()) {
+                verified.getFailureMessages().forEach(monitor::debug);
```

Comment:
log should be more explicit, e.g. `Failure in token verification: [<failure messages>]`

Reply:
Exception messages from nimbus are already pretty detailed, e.g.

`JWT missing required claims: [aud]`
`JWT "aud" claim has value [https://otherserver.com] but should be [test-audience]`

Reply:
Added this prefix